usaspending_api/references/v2/views/autocomplete.py

```diff
@@ -136,23 +136,48 @@ def post(self, request):
 class RecipientAutocompleteViewSet(BaseAutocompleteViewSet):
 
     def post(self, request):
-        """Return all recipients matching the provided search text"""
+        """Return all Parents and Recipients matching the provided search text"""
 
         search_text, limit = self.get_request_payload(request)
 
-        queryset = LegalEntity.objects.all()
+        # Return Recipients with valid id entries
+        queryset = LegalEntity.objects.exclude(
+            recipient_unique_id__isnull=True).exclude(
+            recipient_unique_id__exact='')
 
-        queryset = queryset.annotate(similarity=Greatest(TrigramSimilarity('recipient_name', search_text),
-                                                         TrigramSimilarity('recipient_unique_id', search_text))).\
-            order_by('-similarity')
-
-        exact_match_queryset = queryset.filter(similarity=1.0)
-        if exact_match_queryset.count() > 0:
-            queryset = exact_match_queryset
+        # Filter based on search text
+        response = {}
 
-        column_names = ['legal_entity_id', 'recipient_name', 'recipient_unique_id']
+        # Search and filter for Parent Recipients
+        parents = queryset.annotate(
+            similarity=Greatest(
+                TrigramSimilarity('recipient_name', search_text),
+                TrigramSimilarity('parent_recipient_unique_id', search_text))
+        ).distinct().order_by('-similarity').filter(
+            parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+
+        # Search and filter for Recipients, excluding Parent Recipients
+        recipients = queryset.annotate(
+            similarity=Greatest(
+                TrigramSimilarity('recipient_name', search_text),
+                TrigramSimilarity('recipient_unique_id', search_text))
+        ).distinct().order_by('-similarity').exclude(
+            parent_recipient_unique_id__in=F('recipient_unique_id'))[:limit]
+
+        response['results'] = {
+            'parent_recipient': [
+                parents.values('legal_entity_id',
+                               'recipient_name',
+                               'parent_recipient_unique_id')[:limit]],
+            'recipient': [
+                recipients.values('legal_entity_id',
+                                  'recipient_name',
+                                  'recipient_unique_id')[:limit]]}
+
+        response['counts'] = {'parent_recipient': len(parents),
```

Comment:
I would double check with frontend, but if `counts` isn't actually being used, we shouldn't include it in the response.